pkg/artifactory/resource/security/resource_artifactory_scoped_token.go

```diff
@@ -0,0 +1,288 @@
+package security
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/jfrog/terraform-provider-shared/util"
+	"github.com/jfrog/terraform-provider-shared/validator"
+)
+
+type AccessTokenPostRequest struct {
+	GrantType   string `json:"grant_type"`
+	Username    string `json:"username,omitempty"`
+	Scope       string `json:"scope,omitempty"`
+	ExpiresIn   int    `json:"expires_in,omitempty"`
+	Refreshable bool   `json:"refreshable"`
+	Description string `json:"description"`
+	Audience    string `json:"audience,omitempty"`
+}
+
+type AccessTokenPostResponse struct {
+	TokenId     string `json:"token_id"`
+	AccessToken string `json:"access_token"`
+	ExpiresIn   int    `json:"expires_in"`
+	Scope       string `json:"scope"`
+	TokenType   string `json:"token_type"`
+}
+
+type AccessTokenGet struct {
+	TokenId     string `json:"token_id"`
+	Subject     string `json:"subject"`
+	Expiry      int    `json:"expiry"`
+	IssuedAt    int    `json:"issued_at"`
+	Issuer      string `json:"issuer"`
+	Description string `json:"description"`
+	Refreshable bool   `json:"refreshable"`
+}
+
+func (a AccessTokenPostResponse) Id() string {
+	return a.TokenId
+}
+
+const accessTokensUrl = "access/api/v1/tokens"
```

Comment:
These can also be moved into closure scope - up to you

Reply:
Can the `Id()` be moved inside the function scope? Let me try that.

Reply:
Literally, almost all of it can be - Struct defs, everything